src/Abp/Timing/ZonedDateTimeRange.cs

```diff
@@ -0,0 +1,288 @@
+        {
+            Timezone = timeZoneId;
+        }
+
+        public ZonedDateTimeRange(DateTime startTime, TimeSpan timeSpan, string timeZoneId) : base(startTime, timeSpan)
+        {
+            Timezone = timeZoneId;
+        }
+
+        /// <summary>
+        /// The Timezone of the datetime range
+        /// </summary>
+        public string Timezone { get; set; } = "UTC";
+
+        /// <summary>
+        /// The StartTime with Offset
+        /// </summary>
+        public DateTimeOffset StartTimeOffset
+        {
+            get => TimezoneHelper.ConvertToDateTimeOffset(StartTime, Timezone);
+            set => StartTimeUtc = value.UtcDateTime;
+        }
+
+        /// <summary>
+        /// The EndTime with Offset
+        /// </summary>
+        public DateTimeOffset EndTimeOffset
+        {
+            get => TimezoneHelper.ConvertToDateTimeOffset(EndTime, Timezone);
+            set => EndTimeUtc = value.UtcDateTime;
+        }
+
+        /// <summary>
+        /// The StartTime in UTC
+        /// </summary>
+        public DateTime StartTimeUtc
+        {
+            get => StartTimeOffset.UtcDateTime;
+            set
+            {
+                var localized = TimezoneHelper.ConvertFromUtc(value, Timezone);
+                if (localized.HasValue)
+                {
+                    StartTime = localized.Value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// The EndTime in UTC
+        /// </summary>
+        public DateTime EndTimeUtc
+        {
+            get => EndTimeOffset.UtcDateTime;
+            set
+            {
+                var localized = TimezoneHelper.ConvertFromUtc(value, Timezone);
+                if (localized.HasValue)
+                {
+                    EndTime = localized.Value;
+                }
+            }
+        }
+
+        /// <summary>
+        /// The current time based on the timezone
+        /// </summary>
+        public DateTime Now
+        {
+            get
+            {
+                DateTime? localTime;
+                switch (Clock.Kind)
+                {
+                    case DateTimeKind.Local:
+                        localTime = TimezoneHelper.ConvertFromUtc(Clock.Now.ToUniversalTime(), Timezone);
+                        break;
+                    case DateTimeKind.Unspecified:
+                        localTime = Clock.Now;
+                        break;
+                    case DateTimeKind.Utc:
+                        localTime = TimezoneHelper.ConvertFromUtc(Clock.Now, Timezone);
+                        break;
+                    default:
+                        throw new ArgumentOutOfRangeException();
+                }
+
+                return localTime ?? Clock.Now;
+            }
+        }
+
+        /// <summary>
+        /// Gets a zoned date range representing yesterday.
+        /// </summary>
+        public new ZonedDateTimeRange Yesterday
```

Comment:
Yes, I'm repeating myself here.  Perhaps some DateTimeExtensions could make it DRYer?